packages/components/src/components/domainproperties/dataset/DatasetDesignerPanels.tsx

```diff
@@ -232,44 +289,94 @@ export class DatasetDesignerPanelImpl extends React.PureComponent<Props & Inject
         const { setSubmitting } = this.props;
         const { model, fileImportData } = this.state;
 
-        saveDomain(model.domain, model.getDomainKind(), model.getOptions(), model.name)
-            .then(response => {
-                this.setState(
-                    produce((draft: Draft<State>) => {
-                        const updatedModel = draft.model;
-                        updatedModel.exception = undefined;
-                        updatedModel.domain = response;
-                    }),
-                    () => {
-                        // If we're importing Dataset file data, import file contents
-                        if (fileImportData) {
-                            this.handleFileImport();
-                        } else {
-                            const { model } = this.state;
+        let keyPropIndex;
+        let visitPropIndex;
+
+        if (this._participantId && this._sequenceNum) {
+            // filter out these fields
+            const updatedDomain = model.domain.merge({
+                fields: model.domain.fields
+                    .filter(field => field.name != this._participantId)
+                    .filter(field => field.name != this._sequenceNum)
+                    .toList(),
+            }) as DomainDesign;
+
+            updatedDomain.fields.map((field, index) => {
+                if (model.keyPropertyName && field.name === model.keyPropertyName) {
+                    keyPropIndex = index;
+                }
+                if (model.visitDatePropertyName && field.name === model.visitDatePropertyName) {
+                    visitPropIndex = index;
+                }
+            });
+
+            const updatedModel = produce(model, (draft: Draft<DatasetModel>) => {
+                draft.domain = updatedDomain;
+            });
```

Comment:
I'm not sure we want to be updating the full model domain and keyPropertyIndex / visitPropIndex values in the state at this point since a failure to save the domain would then be messed up in the UI because we've already updated the state object. Instead, I would suggest that we create the `updatedDomain` const like you are with the fields filtered, but then just use that domain in the saveDomain call without updating state at this point.

Reply:
yes, I've changed this in next commit to be only used when there is import data and column mappings are mapped. let me know if it isn't correct in the next commit.